tutorials/physics/using_character_body_2d.rst

```diff
@@ -500,91 +469,57 @@ Here's the code for the player body:
+    var jump_speed = -400.0
 
-    func get_input():
-        velocity.x = 0
-        var right = Input.is_action_pressed('ui_right')
-        var left = Input.is_action_pressed('ui_left')
-        var jump = Input.is_action_just_pressed('ui_select')
+    # Get the gravity from the project settings so you can sync with rigid body nodes.
+    var gravity = ProjectSettings.get_setting("physics/2d/default_gravity")
 
-        if jump and is_on_floor():
-            jumping = true
-            velocity.y = jump_speed
-        if right:
-            velocity.x += run_speed
-        if left:
-            velocity.x -= run_speed
 
     func _physics_process(delta):
-        get_input()
+        # Add the gravity.
         velocity.y += gravity * delta
-        if jumping and is_on_floor():
-            jumping = false
-        velocity = move_and_slide(velocity, Vector2(0, -1))
 
- .. code-tab:: csharp
+        # Handle Jump.
+        if Input.is_action_just_pressed("jump") and is_on_floor():
+            velocity.y = jump_speed
 
-    using Godot;
+        # Get the input direction and handle the movement/deceleration.
+        var direction = Input.get_axis("ui_left", "ui_right")
+        velocity.x = direction * speed
 
-    public partial class KBExample : CharacterBody2D
-    {
-        [Export] public int RunSpeed = 100;
-        [Export] public int JumpSpeed = -400;
-        [Export] public int Gravity = 1200;
+        move_and_slide()
 
-        Vector2 velocity = new Vector2();
-        bool jumping = false;
-
-        public void GetInput()
-        {
-            velocity.x = 0;
-            bool right = Input.IsActionPressed("ui_right");
-            bool left = Input.IsActionPressed("ui_left");
-            bool jump = Input.IsActionPressed("ui_select");
+ .. code-tab:: csharp
 
-            if (jump && IsOnFloor())
-            {
-                jumping = true;
-                velocity.y = JumpSpeed;
-            }
+    using Godot;
 
-            if (right)
-                velocity.x += RunSpeed;
-            if (left)
-                velocity.x -= RunSpeed;
-        }
+    public partial class CBExample : CharacterBody2D
+    {
+        public float Speed = 100.0;
+        public float JumpSpeed = -400.0;
+        public float Gravity = ProjectSettings.GetSetting("physics/2d/default_gravity");
 
-        public override void _PhysicsProcess(float delta)
+        public override void _PhysicsProcess(double delta)
         {
-            GetInput();
-            velocity.y += Gravity * delta;
-            if (jumping && IsOnFloor())
-                jumping = false;
-            velocity = MoveAndSlide(velocity, new Vector2(0, -1));
+            Velocity.y += Gravity * delta;
+            if (Input.IsActionJustPressed("ui_accept") && IsOnFloor())
+                Velocity.y = JumpSpeed;
+            Vector2 direction = Input.GetAxis("ui_left", "ui_right");
+            Velocity.x = direction * Speed;
+            MoveAndSlide();
```

Comment:
Missing comments from the GDScript version. Also, the action seems to be called `jump`. And remember that you can't set the field of a property directly, need to create a separate velocity variable to update X and Y individually, then assign it back to the `Velocity` property of the object.

```suggestion
            Vector2 velocity = Velocity;

            // Add the gravity.
            velocity.y += Gravity * delta;

            // Handle Jump.
            if (Input.IsActionJustPressed("jump") && IsOnFloor())
                velocity.y = JumpSpeed;

            // Get the input direction and handle the movement/deceleration.
            Vector2 direction = Input.GetAxis("ui_left", "ui_right");
            velocity.x = direction * Speed;

            Velocity = velocity;
            MoveAndSlide();
```

Reply:
This is what comes from not using C# regularly. Thanks for the fixes!